python/ray/actor.py

```diff
@@ -715,6 +716,21 @@ def _remote(self, args=None, kwargs=None, **actor_options):
         placement_group_capture_child_tasks = actor_options[
             "placement_group_capture_child_tasks"
         ]
+        if placement_group != "default":
+            warnings.warn(
+                "DeprecationWarning: placement_group parameter is "
+                "deprecated. Use scheduling_strategy parameter instead."
+            )
```

Comment:
Is this how we decide to soft deprecate, instead of doing `warning.warn("deprecation msg", DeprecationWarning), stacklevel=`?

Reply:
Ok, I have seen several ways of deprecations in the codebase (e.g. log.warn(), warnings.warn(), etc). Let's settle down on one.

This is what numpy does
```
warnings.warn('np.asscalar(a) is deprecated since NumPy 1.16.0, use '
              'a.item() instead', DeprecationWarning, stacklevel=3)
 ```
 so we should probably follow the same style. @richardliaw WDYT, since you are working on the deprecation doc.

Reply:
This PR is advocating warnings.warn: https://docs.python.org/2/howto/logging.html#when-to-use-logging

Reply:
Yea, what I'm suggesting is also `warnings.warn`, just the way how we call it.

Reply:
yes, warnings.warn is good. Please make sure you also set the right stacklevel!

Reply:
Is the default stacklevel=1 incorrect here? What should it be?

Reply:
You need to take a look at the stacktrace; maybe 2 or 3. The point is, the user should see the warning rise from THEIR code that they need to change, not Ray internal code.

Reply:
yep, latter

Reply:
Good point, yeah this is case by case and should surface at higher API level.